test/tools/unified-spec-runner/operations.ts

```diff
@@ -377,6 +387,77 @@ operations.set('listIndexes', async ({ entities, operation }) => {
+  const controller = new AbortController();
+  // We always want the process to exit on SIGINT last, so all other
+  // SIGINT events listeners must be prepended.
+  process.prependListener('SIGINT', () => {
+    controller.abort('Process received SIGINT, aborting operation loop.');
+  });
+  const args = operation.arguments!;
+  const storeIterationsAsEntity = args.storeIterationsAsEntity;
+  const storeSuccessesAsEntity = args.storeSuccessesAsEntity;
+  const storeErrorsAsEntity = args.storeErrorsAsEntity;
+  const storeFailuresAsEntity = args.storeFailuresAsEntity;
+
+  if (storeErrorsAsEntity) {
+    entities.set(storeErrorsAsEntity, []);
+  }
+  if (storeFailuresAsEntity) {
+    entities.set(storeFailuresAsEntity, []);
+  }
+
+  let iterations = 0;
+  let successes = 0;
+  while (!controller.signal.aborted) {
+    if (storeIterationsAsEntity) {
+      entities.set(storeIterationsAsEntity, iterations++);
+    }
+    for (const op of args.operations) {
+      try {
+        await executeOperationAndCheck(op, entities, client, testConfig);
+        if (storeSuccessesAsEntity) {
+          entities.set(storeSuccessesAsEntity, successes++);
+        }
+      } catch (error) {
+        // From the unified spec:
+        // If neither storeErrorsAsEntity nor storeFailuresAsEntity are specified,
+        // the loop MUST terminate and raise the error/failure (i.e. the error/failure
+        // will interrupt the test).
+        if (!storeErrorsAsEntity && !storeFailuresAsEntity) {
+          entities.set('errors', [
+            {
+              error: 'Neither storeErrorsAsEntity or storeFailuresAsEntity specified',
+              time: Date.now()
+            }
+          ]);
+          controller.abort('Neither storeErrorsAsEntity or storeFailuresAsEntity specified');
+          return;
+        }
+
+        // From the unified spec format specification for the loop operation:
+        // A failure is when the result or outcome of an operation executed by the test
+        // runner differs from its expected outcome. For example, an expectResult assertion
+        // failing to match a BSON document or an expectError assertion failing to match
+        // an error message would be considered a failure.
+        // An error is any other type of error raised by the test runner. For example, an
+        // unsupported operation or inability to resolve an entity name would be considered
+        // an error.
+        if (storeFailuresAsEntity && error instanceof AssertionError) {
+          entities
+            .getEntity('failures', storeFailuresAsEntity)
+            .push({ error: error.message, time: Date.now() });
+        } else if (storeErrorsAsEntity && !(error instanceof MongoError)) {
+          // Checking not a MongoError ensures it's coming from the test runner.
+          entities
+            .getEntity('errors', storeErrorsAsEntity)
+            .push({ error: error.message, time: Date.now() });
+        }
```

Comment:
optional cleanup: the first condition is redundant
```suggestion
        if (storeFailuresAsEntity) {
       		if (error instanceof AssertionError) {
	          entities
	            .getEntity('failures', storeFailuresAsEntity)
	            .push({ error: error.message, time: Date.now() });
	         }
	         else if (!(error instanceof MongoError)) {
		         entities
	           		.getEntity('errors', storeErrorsAsEntity)
	            	.push({ error: error.message, time: Date.now() });
	         }
        }
```

Reply:
How is the first condition redundant? We only store `failures` when `storeFailuresAsEntity` is defined and only store `errors` when `storeErrorsAsEntity` is defined. The suggestion would fail if `storeErrorsAsEntity` was not defined.

Reply:
Oh, I didn't read it closely enough.